### PR TITLE
python3Packages.scikit-fuzzy: unstable-2022-11-07 -> unstable-2023-09-14

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
@emilazy getting this weird build failure from the CheckPhase
```
============================= test session starts ==============================
platform linux -- Python 3.12.4, pytest-7.4.4, pluggy-1.5.0
rootdir: /build/source
configfile: pyproject.toml
collected 166 items / 1 error                                                  
run-last-failure: None

==================================== ERRORS ====================================
____________ ERROR collecting skfuzzy/cluster/tests/test_cmeans.py _____________
import file mismatch:
imported module 'test_cmeans' has this __file__ attribute:
  /build/source/build/lib/skfuzzy/cluster/tests/test_cmeans.py
which is not the same as the test file we want to collect:
  /build/source/skfuzzy/cluster/tests/test_cmeans.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
=========================== short test summary info ============================
ERROR skfuzzy/cluster/tests/test_cmeans.py
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 1.00s ===============================
```

**Comment by emilazy** [comment]
Looks like some issue with running tests out of tree. It’s beyond my expertise without doing a bunch of digging; hopefully someone else can take a look.

**Review by lavafroth — APPROVED** [review]

**Comment by lavafroth** [comment]
I'll look at the test failures. Looks good otherwise.

**Comment by lavafroth** [comment]
I figured, we gotta remove the build dir during the checkPhase. I added the following:
```nix
  checkPhase = ''
    rm -rf build
  '';
```

File: pkgs/development/python-modules/scikit-fuzzy/default.nix [hunk 1]
```diff
@@ -57,10 +32,11 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     matplotlib
-    nose
     pytest7CheckHook
   ];
 
+  checkPhase = "rm -rf build";
```
Comment (bcdarwin) [inline pkgs/development/python-modules/scikit-fuzzy/default.nix:38]: Shouldn't this be `preCheck`'?

Comment (Sigmanificient) [reply pkgs/development/python-modules/scikit-fuzzy/default.nix]: Whoops indeed, fixed

**Comment by lavafroth** [comment]
I guess that'd be more appropriate, yeah. Also, I just tested that we can run `pytest skfuzzy` in the checkPhase to point to the appropriate directory.

**Comment by bcdarwin** [comment]
> I guess that'd be more appropriate, yeah. Also, I just tested that we can run `pytest skfuzzy` in the checkPhase to point to the appropriate directory.

The current version using `preCheck` is fine, but in the other case `pytestFlagsArray = [ "skfuzzy" ]` would be preferred here.

File: pkgs/development/python-modules/scikit-fuzzy/default.nix [hunk 2]
```diff
@@ -2,51 +2,26 @@
...
 
-buildPythonPackage rec {
+buildPythonPackage {
   pname = "scikit-fuzzy";
-  version = "unstable-2022-11-07";
+  version = "unstable-2023-09-14";
   pyproject = true;
 
-  # code depends on distutils
-  disabled = pythonAtLeast "3.12";
```
Comment (bcdarwin) [inline pkgs/development/python-modules/scikit-fuzzy/default.nix:22]: nit: please set this to `pythonOlder "3.8"`

**Review by bcdarwin — APPROVED** [review]

File: pkgs/development/python-modules/scikit-fuzzy/default.nix [hunk 3]
```diff
@@ -2,51 +2,26 @@
...
   numpy,
   scipy,
   pytest7CheckHook,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage {
   pname = "scikit-fuzzy";
-  version = "unstable-2022-11-07";
+  version = "unstable-2023-09-14";
```
Comment (jopejoe1) [inline pkgs/development/python-modules/scikit-fuzzy/default.nix:15]: When updating please use the new versioning schema for unstable version.

```suggestion
  version = "0.4.2-unstable-2023-09-14";
```

File: pkgs/development/python-modules/scikit-fuzzy/default.nix [hunk 4]
```diff
@@ -1,52 +1,30 @@
...
-buildPythonPackage rec {
+buildPythonPackage {
   pname = "scikit-fuzzy";
-  version = "unstable-2022-11-07";
+  version = "0.4.2-unstable-2023-09-14";
   pyproject = true;
 
-  # code depends on distutils
-  disabled = pythonAtLeast "3.12";
+  disabled = pythonAtLeast "3.9";
```
Comment (emilazy) [inline pkgs/development/python-modules/scikit-fuzzy/default.nix:19]:
```suggestion
  disabled = pythonOlder "3.9";
```

**Review by emilazy — DISMISSED** [review]

**Comment by Sigmanificient** [comment]
done

**Comment by SuperSandro2000** [comment]
@ofborg build python312Packages.scikit-fuzzy python312Packages.intensity-normalization

**Pull request merged**